ambari-server/src/main/python/ambari_server/setupSso.py

```diff
@@ -85,174 +76,111 @@ def validate_options(options):
 
 def populate_sso_provider_url(options, properties):
   if not options.sso_provider_url:
-      provider_url = get_value_from_properties(properties, JWT_AUTH_PROVIDER_URL, JWT_AUTH_PROVIDER_URL_DEFAULT)
-      provider_url = get_validated_string_input("Provider URL [URL] ({0}):".format(provider_url), provider_url, REGEX_URL,
+      provider_url = get_value_from_dictionary(properties, SSO_PROVIDER_URL, SSO_PROVIDER_URL_DEFAULT)
+      provider_url = get_validated_string_input("Provider URL ({0}):".format(provider_url), provider_url, REGEX_URL,
                                                 "Invalid provider URL", False)
   else:
     provider_url = options.sso_provider_url
 
-  properties.process_pair(JWT_AUTH_PROVIDER_URL, provider_url)
-
+  properties[SSO_PROVIDER_URL] = provider_url
 
 def populate_sso_public_cert(options, properties):
   if not options.sso_public_cert_file:
-    cert_path = properties.get_property(JWT_PUBLIC_KEY)
-    cert_string = get_multi_line_input("Public Certificate pem ({0})".format('stored' if cert_path else 'empty'))
-    store_new_cert = False
-    if cert_string is not None:
-        store_new_cert = True
-    if store_new_cert:
-      full_cert = JWT_PUBLIC_KEY_HEADER + cert_string + JWT_PUBLIC_KEY_FOOTER
-      cert_path = store_password_file(full_cert, JWT_PUBLIC_KEY_FILENAME)
+    cert = get_value_from_dictionary(properties, SSO_CERTIFICATE)
+    get_cert = True if not cert else get_YN_input("The SSO provider's public certificate has already set. Do you want to change it [y/n] (n): ", False)
+
+    if get_cert:
+      cert_string = get_multi_line_input("Public Certificate PEM")
+      properties[SSO_CERTIFICATE] = ensure_complete_cert(cert_string) if cert_string else ""
   else:
     cert_path = options.sso_public_cert_file
-
-  properties.process_pair(JWT_PUBLIC_KEY, cert_path)
+    with open(cert_path) as file:
```

Comment:
file is a builtin name. It doesn't cause any problems here in this case but it's good avoid overriding builtins.